django/core/checks/security/base.py

```diff
@@ -119,6 +119,11 @@
     id='security.E023',
 )
 
+W024 = Warning(
```

Comment:
I feel like this should be an Error and not a Warning. My understanding is that a Warning is for something that is strongly recommended, but things will hopefully still more-or-less work. Whereas an Error, you can be pretty certain things won't work. 

If you're `ALLOWED_HOSTS` is set to a string you won't even be able to accept a request. That puts this firmly in the Error camp.

Reply:
Yeah Error does make sense, i'll move this to E024

Reply:
Updated in https://github.com/django/django/pull/14149/commits/1b9132f81eca543877ed5d0479cf65066ad154c5